File: src/HotelOrderSystem.Api/Dtos/OrderDtos.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json;

namespace HotelOrderSystem.Api.Dtos;

public sealed record CreateOrderRequest(
    [property: Range(1, int.MaxValue)] int RoomId,
    [property: MaxLength(50)] string Source,
    [property: Required, MinLength(1)] List<CreateOrderLineRequest> Items);

public sealed record CreateOrderLineRequest(
    [property: Range(1, int.MaxValue)] int ItemId,
    [property: Range(1, int.MaxValue)] int Quantity,
    JsonElement? DynamicAttributes);

public sealed record CreateGuestOrderRequest(
    [property: Required, MinLength(1)] List<CreateOrderLineRequest> Items);

public sealed record AcceptOrderRequest(string? RowVersion);

public sealed record CompleteOrderRequest(string? Notes);

public sealed record CancelOrderRequest(string? Reason);

public sealed record OrderDto(
    int OrderId,
    int RoomId,
    string RoomNumber,
    int? AssignedTeamId,
    string? AssignedTeamName,
    string Source,
    string Status,
    int? CreatedByUserId,
    string? CreatedByUserName,
    int? AcceptedByUserId,
    string? AcceptedByUserName,
    DateTime CreatedAtUtc,
    DateTime? AcceptedAtUtc,
    DateTime? CompletedAtUtc,
    DateTime? SlaDueAtUtc,
    DateTime? EscalatedAtUtc,
    string RowVersion,
    List<OrderDetailDto> Details);

public sealed record OrderDetailDto(
    int OrderDetailId,
    int ItemId,
    string ItemName,
    int Quantity,
    string DynamicAttributes);

public sealed record CreateOrderResponse(IReadOnlyList<OrderDto> Orders);

File: android-staff-app/app/src/main/java/com/ibaapps/HotelOrderSystem/data/remote/dto/OrderDtos.kt
package com.ibaapps.HotelOrderSystem.data.remote.dto

import kotlinx.serialization.Serializable

@Serializable
data class OrderDto(
    val orderId: Int,
    val roomId: Int,
    val roomNumber: String,
    val assignedTeamId: Int? = null,
    val assignedTeamName: String? = null,
    val source: String,
    val status: String,
    val createdByUserId: Int? = null,
    val createdByUserName: String? = null,
    val acceptedByUserId: Int? = null,
    val acceptedByUserName: String? = null,
    val createdAtUtc: String,
    val acceptedAtUtc: String? = null,
    val completedAtUtc: String? = null,
    val slaDueAtUtc: String? = null,
    val escalatedAtUtc: String? = null,
    val rowVersion: String,
    val details: List<OrderDetailDto> = emptyList()
)

@Serializable
data class OrderDetailDto(
    val orderDetailId: Int,
    val itemId: Int,
    val itemName: String,
    val quantity: Int,
    val dynamicAttributes: String = "{}"
)

@Serializable
data class AcceptOrderRequestDto(
    val rowVersion: String? = null
)

@Serializable
data class CompleteOrderRequestDto(
    val notes: String? = null
)

@Serializable
data class CancelOrderRequestDto(
    val reason: String? = null
)
